src/docfx/build/page/BookmarkValidator.cs

```diff
@@ -59,9 +59,7 @@ public void Validate()
                 if (bookmarks.Contains(bookmark))
                     continue;
 
-                var error = isSelfBookmark
-                    ? Errors.InternalBookmarkNotFound(source, file, bookmark, bookmarks)
-                    : Errors.ExternalBookmarkNotFound(source, reference, bookmark, bookmarks);
+                var error = Errors.BookmarkNotFound(source, isSelfBookmark ? file : reference, bookmark, bookmarks);
```

Comment:
remove isSelfBookmark here and in line 25. In `DependencyResolver.cs` line 82, pass `(declaringFile, file, fragment, path)`, also remove `// NOTE: bookmark validation result depend on current inclusion stack`

```suggestion
                var error = Errors.BookmarkNotFound(source, reference, bookmark, bookmarks);
```